src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -3691,4 +3774,24 @@ private static string GetCompactJson(TestCaseType testCaseType, string jsonStrin
             return s_compactJson[testCaseType] = existing;
         }
     }
+    public class ThrowOnReadStream : MemoryStream
+    {
+        public ThrowOnReadStream(byte[] bytes) : base(bytes)
+        {
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    public class ThrowOnSeekStream : MemoryStream
+    {
+        public ThrowOnSeekStream(byte[] bytes) : base(bytes)
+        {
+        }
+
+        public override bool CanSeek => throw new NotImplementedException();
```

Comment:
A more unique exception would make the test more identifiable.  Something that isn't generally thrown, such as ApplicationException.  (Then the calling/testing code looks less like we didn't implement JsonDocument.Parse).

Reply:
For me NotImplementedException makes more sense than ApplicationException as that is the most common excepotion could happen/thrown for CanSeek property
> (Then the calling/testing code looks less like we didn't implement JsonDocument.Parse)

Sorry didn't understood this point well. Anyways the test intended for code coverage, not about if JsonDocument.Parse is implemented, but guess it is not what you meant 😕

Reply:
The Assert line in your test is `Assert.Throws<NotImplementedException>(() => JsonDocument.Parse(stream))`.  That looks, at a casual glance, like JsonDocument.Parse is what is throwing the exception; but you're adding coverage for a broken stream implementation.

While NotImplementedException is the most likely exception (since auto-generated code tends to leave that sprinkled about), it's not handled special by the calling code.  So I would use some other exception type, just to indicate that it's an exception from the Stream implementation instead of from the JsonDocument implementation.  It's not a major issue, though.

Reply:
Got it thanks, updated with more meaningful exceptions